pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/kafka/MessageBatch.java

```diff
@@ -54,5 +54,5 @@
      * @param index
      * @return
      */
-    long getNextKafkaMessageOffsetAtIndex(int index);
+    long getNextKafkaMessageOffsetAtIndex(int index, long lastOffsetPosition);
```

Comment:
you should not need lastOffsetposition as an arg. just construct the messagebatch with this position